apps/api/src/app/events/usecases/send-message/send-message-email.usecase.ts

```diff
@@ -350,14 +357,12 @@ export class SendMessageEmail extends SendMessageBase {
     mailData: IEmailOptions,
     message: MessageEntity,
     command: SendMessageCommand,
-    notification: NotificationEntity
+    notification: NotificationEntity,
+    senderName?: string
   ) {
     const mailFactory = new MailFactory();
     const mailHandler = mailFactory.getHandler(
-      {
-        ...integration,
-        providerId: GetNovuIntegration.mapProviders(ChannelTypeEnum.EMAIL, integration.providerId),
-      },
+      SendMessageEmail.buildFactoryIntegration(integration, senderName),
```

Comment:
What's the goal of using a static method rather than the instance method considering where it is called it will be from a method of an instantiated class?

Reply:
No one really... I was thinking that we might reuse it in other instances but figured out now it is email specific...